lib/daru/vector.rb

```diff
@@ -879,41 +879,114 @@ def to_s
+                "\n  Distribution"
+
+      key, val, dist = frequencies.sort.each_with_index.map do |v, k|
+        [k, v, '%0.2f%%' % ((nval.zero? ? 1 : v.quo(nval))*100)]
+      end.transpose
+
+      summary << self.class.table([key, val, dist])
+      summary
+    end
+
+    # Displays summary for an numeric type Vector
+    # @return [String] String containing numeric vector summary
+    def numeric_summary
+      summary = "\n  median: #{median}" +
+                "\n  mean: %0.4f" % mean
+      if sd
+        summary << "\n  std.dev.: %0.4f" % sd +
+                   "\n  std.err.: %0.4f" % se +
+                   "\n  skew: %0.4f" % skew +
+                   "\n  kurtosis: %0.4f" % kurtosis
+      end
+      summary
     end
 
-    # :nocov:
-    def report_building b # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
-      b.section(name: name) do |s|
-        s.text "n :#{size}"
-        s.text "n valid:#{count_values(*Daru::MISSING_VALUES)}"
-        if @type == :object
-          s.text  "factors: #{factors.to_a.join(',')}"
-          s.text  "mode: #{mode}"
-
-          s.table(name: 'Distribution') do |t|
-            frequencies.sort_by(&:to_s).each do |k,v|
-              key = @index.include?(k) ? @index[k] : k
-              t.row [key, v, ('%0.2f%%' % (v.quo(count_values(*Daru::MISSING_VALUES))*100))]
-            end
-          end
+    # Display summary in Table format
+    # @params [Array] Array of arrays containing values of each column.
+    #                 Each column should have equal number of elements
+    # @return [String] String displaying the summary in table format
+    # @example
+    #   cols=[[1,2,3], [2,3,4]]
+    #   puts Daru::Vector.table(cols)
+    #
+    #   +---+---+
+    #   | 1 | 2 |
+    #   | 2 | 3 |
+    #   | 3 | 4 |
+    #   +---+---+
+
+    class << self
+      def table(cols=[])
+        max_length = max_col_length(cols)
+        table_border(max_length) + table_body(cols) + table_border(max_length)
+      end
+
+      private
+
+      def max_col_length(cols=[])
+        max_length = cols.map { |col| col.map { |e| e.to_s.length }.max }
+        max_length
+      end
+
+      def table_border(max_col_length=[])
+        border = "\n+"
+        max_col_length.each do |len|
+          border << '-' * (len+2) + '+'
         end
+        border
+      end
 
-        s.text "median: #{median}" if @type==:numeric || @type==:numeric
-        if @type==:numeric
-          s.text 'mean: %0.4f' % mean
-          if sd
-            s.text 'std.dev.: %0.4f' % sd
-            s.text 'std.err.: %0.4f' % se
-            s.text 'skew: %0.4f' % skew
-            s.text 'kurtosis: %0.4f' % kurtosis
+      def table_body(cols=[])
+        body = ''
+        max_length = max_col_length(cols)
+        cols[0].each_index do |i|
+          body << "\n| "
+          cols.each_index do |j|
+            body << "#{cols[j][i].to_s.ljust(max_length[j])} | "
           end
         end
+        body
       end
```

Comment:
Looks like there is no point having those methods in `Vector` class, they do nothing with concept of vector by themselves. We have `daru/formatters/` forlder for this kind of things... And even `daru/formatters/table.rb` there. I am not sure it suits your needs, but at least all dedicated formatting code should go there, not be mixed up in any class that needs fancy formatting.